.github/workflows/integration.yml

```diff
@@ -126,40 +117,73 @@ jobs:
           source ~/.poetry/env
           poetry run coverage report
           poetry run coverage xml -o coverage-reports/coverage-report.xml
-
+      - name: Archive coverage report
+        uses: actions/upload-artifact@v2
+        with:
+          name: api-coverage-report
+          path: ./api/coverage-reports/coverage-report.xml
+  lint-and-test-web:
+    name: Web - Lint, Test with coverage
+    runs-on: ubuntu-20.04
+    strategy:
+      matrix:
+        node-version: [14.x]
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v2
+        with:
+          # For sonar-scanner to work properly we can't use a shallow fetch.
+          fetch-depth: 0
+      - name: Setup kernel for react, increase watchers
+        run: echo fs.inotify.max_user_watches=524288 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v1
+        with:
+          node-version: ${{ matrix.node-version }}
       - uses: actions/cache@v2
         with:
           path: |
             **/node_modules
             ~/.cache/Cypress
           key: ${{ runner.os }}-modules-${{ hashFiles('**/yarn.lock') }}
-
       - name: Install node dependencies (web)
         working-directory: ./web
         if: steps.yarn-cache.outputs.cache-hit != 'true'
         run: yarn install
-
       - name: Lint (web)
         working-directory: ./web
         run: yarn run lint
-
       # "Error: ENOSPC: System limit for number of file watchers reached" can be addressed
       # with this: https://github.com/guard/listen/wiki/Increasing-the-amount-of-inotify-watchers#the-technical-details
       # It seems unnecessary at the moment because tests pass anyway
       - name: Cypress tests with coverage (web)
         working-directory: ./web
         run: yarn run cypress:ci
-
       - name: Unit tests (web)
         working-directory: ./web
         run: yarn run coverage:ci
-
       - name: Merge and finalize test coverage (web)
         working-directory: ./web
         run: yarn run finalizeCoverage
 
-      # - name: Upload test coverage to Codecov
-      #   uses: codecov/codecov-action@v1.5.2
-      #   with:
-      #     token: ${{secrets.CODECOV_TOKEN}}
-      #     fail_ci_if_error: false
+      - name: Archive coverage report
+        uses: actions/upload-artifact@v2
+        with:
+          name: web-coverage-report
+          path: ./web/finalCoverage
+  upload-code-coverage:
+    name: Coverage with CodeCov
+    runs-on: ubuntu-20.04
+    needs: [lint-and-test-web, lint-and-test-api]
+    steps:
+      # we need to checkout, so that we have codecov.yml
+      - name: Checkout repo
+        uses: actions/checkout@v2
+      - name: Download all workflow run artifacts
```

Comment:
Was it possible to verify that all coverage reports were downloaded and posted to CodeCov?